### ISSUE TITLE
[Feature]: Please Add Parameterized Timeout Setting

### ISSUE THREAD

**Issue by kbpitula** [IBMa/equal-access#2036]
### Project

accessibility-checker for Node

### Description

I am using the a11y checker as part of my Selenium UI tests. When I execute the scan (e.g., aChecker.getCompliance(driver, label)), it fails with the error: ScriptTimeoutError: script timeout.

I tried increasing the WebDriver timeout (e.g., await driver.manage().setTimeouts({ script: 5000000 });), but the issue persists.

I investigated where the a11y-checker sets the timeout and made some modifications to check for the problem. I changed all timeouts in ACHelper and ACEngineManager:

From 60000 to 10000000
From 30000 to 10000000

The script then generated the report.

Is it possible to parameterize this timeout setting so that I can adjust it for my script?

**Comment by shunguoy** [comment]
https://ibm-studios.slack.com/archives/C036P1CTN/p1726150186615779

**Comment by ianwhale** [comment]
@kbpitula Could you share where you changed these values to increase the timeout?

I'm having the same issue.

**Comment by kbpitula** [comment]
@ianwhale 
I am using this command in automation:

```
sed -i -e 's/60000/'10000000'/' node_modules/accessibility-checker/lib/*.js
sed -i -e 's/300000/'10000000'/' node_modules/accessibility-checker/lib/*.js

```

**Comment by ianwhale** [comment]
@kbpitula Thanks for the quick reply!

Its been a while since I've had to use javascript 😬 

Is there something else I need to do after editing the `.js` files in `node_modules`? I can see the change in my files but when in run I still get: 

```
Navigation timeout of 30000 ms exceeded file://<path>/index.html
```

**Comment by kbpitula** [comment]
@ianwhale I updated these values in node_modules/accessibility-checker/lib/*.js

After that the problem not occurs for me.

**Comment by ianwhale** [comment]
@kbpitula Got it! Thanks.

**Comment by ianwhale** [comment]
For posterity's sake, I ended up fixing this by directly passing a `puppeteer` object to `aChecker.getCompliance` rather than the file path itself.

That way one can set the timeout and navigate with `puppeteer` rather than relying on the `aChecker` object to do it.

For example:

```javascript
const browser = await puppeteer.launch({ headless: "true" });
const page = await browser.newPage();
page.setDefaultNavigationTimeout(0);
await page.goto("file://index.html", { waitUntil: 'load' });
const result = await aChecker.getCompliance(page, "index.html");
```
